pithos/plugins/notification_icon.py

```diff
@@ -83,31 +145,32 @@ def button(text, action, checked=False):
             menu.append(item)
             return item, handler
 
-        if indicator_capable:
+        if self.appindicator:
             # We have to add another entry for show / hide Pithos window
-            self.visible_check, handler = button("Show Pithos", self._toggle_visible, True)
+            self.visible_check, handler = button('Show Pithos', self._toggle_visible, True)
 
             def set_active(active):
                 GObject.signal_handler_block(self.visible_check, handler)
                 self.visible_check.set_active(active)
                 GObject.signal_handler_unblock(self.visible_check, handler)
+                return True
 
             # Ensure it is kept in sync
-            self.window.connect("hide", lambda w: set_active(False))
-            self.window.connect("show", lambda w: set_active(True))
+            self.window.connect('hide', lambda w: set_active(False))
+            self.window.connect('show', lambda w: set_active(True))
 
             # On middle-click
             self.ind.set_secondary_activate_target(self.visible_check)
 
-        self.playpausebtn = button("Pause", self.window.playpause)[0]
-        button("Skip", self.window.next_song)
-        button("Love", (lambda *i: self.window.love_song()))
-        button("Ban", (lambda *i: self.window.ban_song()))
-        button("Tired", (lambda *i: self.window.tired_song()))
-        button("Quit", self.window.quit)
+        self.playpausebtn = button('Pause', self.window.playpause)[0]
```

Comment:
> I forget if I already asked, but does Plasma not own org.kde.StatusNotifierWatcher ?

appindicator is basically a sh!t wrapper around a DBus service kinda like libnotify except ofc it's the creator not the consumer. KDE/Qt tray icons work in Unity with little to no modification without the aid of appindicator if I remember correctly.